src/libcore/convert.rs

```diff
@@ -552,7 +550,12 @@ impl<T> From<T> for T {
 
 #[stable(feature = "convert_infallible", since = "1.34.0")]
 #[cfg(not(bootstrap))]
-#[rustc_reservation_impl]
+#[rustc_reservation_impl="a future version of Rust might implement `From<!>` for \
+    all types. \
+    However, it is OK to implement `From<!>` for types you own - \
```

Comment:
The complexity of this error message and describing the situation to users pushes me further in the direction of thinking that we should go ahead and provide the `impl<T> From<!> for T`.